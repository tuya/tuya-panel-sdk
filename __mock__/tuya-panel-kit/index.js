import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
