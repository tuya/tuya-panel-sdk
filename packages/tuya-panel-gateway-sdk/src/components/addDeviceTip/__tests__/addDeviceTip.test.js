import React from 'react';
import { Divider, TYSdk } from 'tuya-panel-kit';
import { shallow } from 'enzyme';
import { AddDeviceTip, AddDeviceTipModal } from '../index';
const dataSource = [
  {
    name: 'Sensor',
    icon: 'https://images.tuyacn.com/smart/program_category_icon/zig_afcg.png',
    content: 'Power on, then hold the RESET button for 5 sec',
  },
  {
    name: 'Socket',
    icon: 'https://images.tuyacn.com/smart/product_icon2/cz_1.png',
    content: 'Power on, then hold the RESET button for 5 secç§’',
  },
  {
    name: 'Light Source',
    icon: 'https://images.tuyacn.com/smart/product_icon2/dj_1.png',
    content: 'Power On, then Turn OFF-ON-OFF-ON',
  },
];
describe('AddDeviceTip components', () => {
  it('basic render', () => {
    const wrapper = shallow(<AddDeviceTip dataSource={dataSource} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('change title and description', () => {
    const wrapper = shallow(
      <AddDeviceTip title="Distribution network prompt" desc="Distribution network description" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('change button style', () => {
    const wrapper = shallow(
      <AddDeviceTip
        addButtonStyle={{
          backgroundColor: 'lightblue',
        }}
        addButtonTextStyle={{
          color: '#333',
        }}
        moreButtonTextStyle={{
          color: 'red',
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('change divider color', () => {
    const wrapper = shallow(<AddDeviceTip dividerColor="lightblue" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('change content padding horizontal', () => {
    const wrapper = shallow(<AddDeviceTip contentPaddingHorizontal={60} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('change numberOfLines', () => {
    const wrapper = shallow(
      <AddDeviceTip titleNumberOfLines={2} descNumberOfLines={2} itemNumberOfLines={1} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render SeparatorComponent', () => {
    const wrapper = shallow(
      <AddDeviceTip renderSeparatorComponent={() => <Divider color="pink" height={2} />} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('button onPress event', () => {
    const wrapper = shallow(
      <AddDeviceTip
        moreButtonOnPress={() => TYSdk.mobile.simpleTipDialog('more button onPress', () => {})}
        addButtonOnPress={() => TYSdk.mobile.simpleTipDialog('add button onPress', () => {})}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
