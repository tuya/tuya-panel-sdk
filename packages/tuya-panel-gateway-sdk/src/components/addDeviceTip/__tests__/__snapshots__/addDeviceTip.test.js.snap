// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDeviceTip components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "content": "Power on, then hold the RESET button for 5 sec",
            "icon": 1,
            "name": "Sensor",
          },
          Object {
            "content": "Power on, then hold the RESET button for 5 secç§’",
            "icon": 1,
            "name": "Socket",
          },
          Object {
            "content": "Power On, then Turn OFF-ON-OFF-ON",
            "icon": 1,
            "name": "Light Source",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components button onPress event 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components change button style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {
            "color": "red",
          },
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {
            "backgroundColor": "lightblue",
          },
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "color": "#333",
          },
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components change content padding horizontal 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 60,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 630,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components change divider color 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="lightblue"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components change numberOfLines 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={2}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={2}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components change title and description 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Distribution network prompt
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Distribution network description
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`AddDeviceTip components render SeparatorComponent 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 32,
          "paddingTop": 72,
        },
        Object {
          "width": 710,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#22242C",
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 28,
            "marginBottom": 20,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      Please make sure the child devices are already in distribution network model (lights flashing)
    </TYText>
    <TYText
      align={null}
      color={null}
      numberOfLines={3}
      size={null}
      style={
        Array [
          Object {
            "color": "#495054",
            "fontSize": 15,
            "lineHeight": 21,
            "paddingHorizontal": 52,
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      If the light is not flashing equipment, please reset the device, the following are some common subset of reset:
    </TYText>
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flexGrow": 0,
            "marginTop": 52,
            "paddingHorizontal": 64,
          },
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginTop": 32,
            "paddingBottom": 20,
            "width": 500,
          },
          Object {},
        ]
      }
      text="See more device reset way >"
      textStyle={
        Array [
          Object {
            "color": "#0091FF",
            "fontSize": 14,
          },
          Object {},
        ]
      }
      theme={Object {}}
    />
    <Divider
      color="#E8E8E8"
      flexDirection="row"
      height={1}
      style={null}
      visible={true}
      width={null}
    />
    <ThemedButton
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF4800",
            "borderRadius": 48,
            "height": 94,
            "overflow": "hidden",
            "width": 440,
          },
          Object {},
        ]
      }
      text="Light is in a flash"
      textStyle={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
      theme={Object {}}
      wrapperStyle={
        Array [
          Object {
            "alignItems": "center",
            "height": 174,
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    />
  </View>
  <ThemedButton
    icon="close"
    iconColor="#FFF"
    onPress={[Function]}
    size={16}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 33,
          "height": 66,
          "overflow": "hidden",
          "width": 66,
        },
        Object {},
      ]
    }
    textStyle={
      Array [
        Object {},
      ]
    }
    theme={Object {}}
    wrapperStyle={
      Array [
        Object {
          "position": "absolute",
          "right": 48,
          "top": -88,
        },
        Object {},
      ]
    }
  />
</View>
`;
